corgidrp/astrom.py

```diff
@@ -0,0 +1,702 @@
+
+    # use only center quadrant
+    imageshape = np.shape(image)
+    quady, quadx = imageshape[0] // 4, imageshape[1] // 4
+    center_source_inds = np.where((guesses['x'] >= quadx) & (guesses['x'] <= imageshape[1] - quadx) & (guesses['y'] >= quady) & (guesses['y'] <= imageshape[0] - quady))
+    quad_guesses = guesses[center_source_inds]
+
+    # create the predicted image header from found platescale and north angle
+    vert_ang = np.radians(cal_properties[1])
+    pc = np.array([[-np.cos(vert_ang), np.sin(vert_ang)], [np.sin(vert_ang), np.cos(vert_ang)]])
+    cdmatrix = pc * (cal_properties[0] * 0.001) / 3600.
+
+    new_hdr = {}
+    new_hdr['CD1_1'] = cdmatrix[0,0]
+    new_hdr['CD1_2'] = cdmatrix[0,1]
+    new_hdr['CD2_1'] = cdmatrix[1,0]
+    new_hdr['CD2_2'] = cdmatrix[1,1]
+    new_hdr['CRPIX1'] = np.shape(image)[1] // 2
+    new_hdr['CRPIX2'] = np.shape(image)[0] // 2
+    new_hdr['CTYPE1'] = 'RA---TAN'
+    new_hdr['CTYPE2'] = 'DEC--TAN'
+    new_hdr['CDELT1'] = (cal_properties[0] * 0.001) / 3600
+    new_hdr['CDELT2'] = (cal_properties[0] * 0.001) / 3600
+    new_hdr['CRVAL1'] = target_coordinate[0]
+    new_hdr['CRVAL2'] = target_coordinate[1]
+    w = astropy.wcs.WCS(new_hdr)
+
+    x_sky_to_pix, y_sky_to_pix = astropy.wcs.utils.skycoord_to_pixel(skycoords, wcs=w)
+    x_predict, y_predict = x_sky_to_pix[center_source_inds], y_sky_to_pix[center_source_inds]
+
+    # find offset between measured centers and predicted positions    
+    image_centerings = np.zeros((len(quad_guesses), 2))
+    boresights = np.zeros((len(quad_guesses), 2))
+    searchrad = 5
+    for i, (xg, yg) in enumerate(zip(quad_guesses['x'], quad_guesses['y'])):
+        p, f, xi_center, yi_center = fakes.gaussfit2d(frame= image, xguess= xg, yguess= yg, searchrad=searchrad)
+        x_off = xi_center - x_predict[i]
+        y_off = yi_center - y_predict[i]
+        boresights[i,:] = [x_off, y_off]
+        image_centerings[i,:] = [xi_center, yi_center]
+
+    # average all offsets in x,y directions
+    boresight_x, boresight_y = np.mean(boresights[:,0]), np.mean(boresights[:,1])
+
+    # convert back to RA, DEC
+    image_center_RA = target_coordinate[0] - ((boresight_x * cal_properties[0]) * astropy.units.mas).to(astropy.units.deg).value
+    image_center_DEC = target_coordinate[1] - ((boresight_y * cal_properties[0]) * astropy.units.mas).to(astropy.units.deg).value
+
+    return image_center_RA, image_center_DEC
+
+def boresight_calibration(input_dataset, field_path, threshold=100, sigma=3, mask_rad=1, comparison_threshold=25):
+    """
+    Perform the boresight calibration of a dataset.
+    
+    Args:
+        input_dataset (corgidrp.data.Dataset): Dataset containing a single image for astrometric calibration
+        field_path (str): Full path to directory with search field data (ra, dec, vmag, etc.)
+        threshold (int): Number of stars to find (default 25)
+        sigma (int): Stddev of point source psf (default: 3 pixels)
```

Comment:
Could we input the `fwhm` instead of `sigma`. We usually remember the value for FWHM more easily, and it is a more direct connection to the `mask_rad` keyword.